[RDS] RDSv3 `read-replica` panics

# Affected Resource(s)
`resource/opentelekomcloud_rds_read_replica_v3`

# Debug Output/Panic Output
```
=== RUN   TestAccRdsReadReplicaV3Encrypted
panic: runtime error: index out of range [0] with length 0

goroutine 865 [running]:
github.com/opentelekomcloud/terraform-provider-opentelekomcloud/opentelekomcloud/services/rds.getReplicaPublicIP(0xc0007f8400)
	C:/Users/rodgyrbu/Documents/terraform-provider-opentelekomcloud/opentelekomcloud/services/rds/resource_opentelekomcloud_rds_read_replica_v3.go:196 +0xda
github.com/opentelekomcloud/terraform-provider-opentelekomcloud/opentelekomcloud/services/rds.resourceRdsReadReplicaV3Create({0x254b478, 0xc000e865a0}, 0xc0007f8400, {0x2317440, 0xc000df0380})
```

--- a/opentelekomcloud/acceptance/rds/resource_opentelekomcloud_rds_read_replica_v3_test.go
+++ b/opentelekomcloud/acceptance/rds/resource_opentelekomcloud_rds_read_replica_v3_test.go
@@ -29,6 +29,15 @@ func TestAccRdsReadReplicaV3Basic(t *testing.T) {
 		CheckDestroy:      testAccCheckRdsInstanceV3Destroy,
 		Steps: []resource.TestStep{
 			{
+				Config: testAccRdsReadReplicaV3BasicNoIP(postfix),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRdsInstanceV3Exists(resName, &rdsInstance),
+					resource.TestCheckResourceAttr(resName, "availability_zone", secondAZ),
+					resource.TestCheckResourceAttr(resName, "volume.0.size", "40"),
+					resource.TestCheckResourceAttr(resName, "public_ips.#", "0"),
+				),
+			},
+			{
 				Config: testAccRdsReadReplicaV3Basic(postfix),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckRdsInstanceV3Exists(resName, &rdsInstance),
@@ -43,15 +52,6 @@ func TestAccRdsReadReplicaV3Basic(t *testing.T) {
 					testAccCheckRdsInstanceV3Exists(resName, &rdsInstance),
 					resource.TestCheckResourceAttr(resName, "availability_zone", secondAZ),
 					resource.TestCheckResourceAttr(resName, "volume.0.size", "40"),
-				),
-			},
-			{ // and assign it back
-				Config: testAccRdsReadReplicaV3Basic(postfix),
-				Check: resource.ComposeTestCheckFunc(
-					testAccCheckRdsInstanceV3Exists(resName, &rdsInstance),
-					resource.TestCheckResourceAttr(resName, "availability_zone", secondAZ),
-					resource.TestCheckResourceAttr(resName, "volume.0.size", "40"),
-					resource.TestCheckResourceAttr(resName, "public_ips.#", "1"),
 				),
 			},
 		},

--- a/opentelekomcloud/services/rds/resource_opentelekomcloud_rds_read_replica_v3.go
+++ b/opentelekomcloud/services/rds/resource_opentelekomcloud_rds_read_replica_v3.go
@@ -193,7 +193,11 @@ func resourceRdsReadReplicaV3Create(ctx context.Context, d *schema.ResourceData,
 }
 
 func getReplicaPublicIP(d *schema.ResourceData) string {
-	return d.Get("public_ips").(*schema.Set).List()[0].(string)
+	ips := d.Get("public_ips").(*schema.Set)
+	if ips.Len() == 0 {
+		return ""
+	}
+	return ips.List()[0].(string)
 }
 
 func getReplicaPrivateIP(d *schema.ResourceData) string {

--- a/releasenotes/notes/fix-rds-rr-659c927b3edd5907.yaml
+++ b/releasenotes/notes/fix-rds-rr-659c927b3edd5907.yaml
@@ -1,0 +1,4 @@
+---
+fixes:
+  - |
+    **[RDS]** Fix panic when creating ``resource/opentelekomcloud_rds_read_replica_v3`` with empty ``public_ips`` (`#1398 <https://github.com/opentelekomcloud/terraform-provider-opentelekomcloud/pull/1398>`_)

